core/src/main/java/com/spotify/metrics/core/SemanticMetricBuilderFactory.java

```diff
@@ -0,0 +1,47 @@
+package com.spotify.metrics.core;
+
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.Reservoir;
+import com.codahale.metrics.Timer;
+import java.lang.reflect.InvocationTargetException;
+import java.util.function.Supplier;
+
+public class SemanticMetricBuilderFactory {
+
+    public static SemanticMetricBuilder<Timer> timerWithReservoir(
+        final Supplier<Reservoir> reservoirSupplier) {
+        return reservoir(Timer.class, reservoirSupplier);
+    }
+
+    public static SemanticMetricBuilder<Histogram> histogramWithReservoir(
+        final Supplier<Reservoir> reservoirSupplier) {
+        return reservoir(Histogram.class, reservoirSupplier);
+    }
+
+    private static <T extends Metric> SemanticMetricBuilder<T>  reservoir(Class<T> klazz,
+        final Supplier<Reservoir> reservoirSupplier) {
+        return new SemanticMetricBuilder<T> () {
+            @Override
+            public T newMetric() {
+                try {
+                    return klazz.getConstructor(Reservoir.class).newInstance(reservoirSupplier.get());
+                } catch (NoSuchMethodException e) {
+                    e.printStackTrace();
+                } catch (IllegalAccessException e) {
+                    e.printStackTrace();
+                } catch (InstantiationException e) {
+                    e.printStackTrace();
+                } catch (InvocationTargetException e) {
+                    e.printStackTrace();
+                }
+                return null;
```

Comment:
should just throw a runtime expectation since we cannot really recover from this... unless we just fall back to the `defaultReservoir`?